[icu] Regression - build fails for x64-linux-release

<!-- ⚠ Please use the generated issue template instead to report a port build failure. ⚠ -->
<!-- ⁉ You can find its location at the end of the console output. ⁉ -->

**Host Environment**

- OS: Docker, debian:11-slim
- Compiler: clang (current from apt)

**To Reproduce**

Steps to reproduce the behavior:

`vcpkg --triplet x64-linux-release install icu`

**Failure logs**

```
-- Installing x64-linux-release-rel
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:36 (message):
  Couldn't find tool "icupkg":

      "/opt/vcpkg/packages/icu_x64-linux-release/tools/icu/sbin/icupkg" does not exist
Call Stack (most recent call first):
  ports/icu/portfile.cmake:162 (vcpkg_copy_tools)
  scripts/ports.cmake:147 (include)

    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: b4f29c544 2023-01-30 (11 hours ago)
```

- Cut and paste the appropriate build messages from the console output.
- Attach ***every failure log file*** mentioned in the console output.

<!-- ‼ **The console output by itself is not enough to investigate a build failure.** ‼ -->

**Additional context**

Broken by #29196 / cdbcc0321216fe46e4b20941f76ad21429d2a90c @m-kuhn 

The `x64-linux-release` is a "community triplet" but it is very useful to speed up automatic builds and CI.

--- a/ports/icu/portfile.cmake
+++ b/ports/icu/portfile.cmake
@@ -158,7 +158,7 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
 endif()
 
 # Install executables from /tools/icu/sbin to /tools/icu/bin on unix (/bin because icu require this for cross compiling)
-if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
+if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX AND "tools" IN_LIST FEATURES)
     vcpkg_copy_tools(
         TOOL_NAMES icupkg gennorm2 gencmn genccode gensprep
         SEARCH_DIR "${CURRENT_PACKAGES_DIR}/tools/icu/sbin"

--- a/ports/icu/vcpkg.json
+++ b/ports/icu/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "icu",
   "version": "72.1",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Mature and widely used Unicode and localization library.",
   "homepage": "https://icu.unicode.org/home",
   "license": "ICU",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3022,7 +3022,7 @@
     },
     "icu": {
       "baseline": "72.1",
-      "port-version": 1
+      "port-version": 2
     },
     "ideviceinstaller": {
       "baseline": "1.1.2.23",

--- a/versions/i-/icu.json
+++ b/versions/i-/icu.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c06697d7c811e568d2e520db13e41fe656bb5468",
+      "version": "72.1",
+      "port-version": 2
+    },
+    {
       "git-tree": "2a4109603660116f3cb7477c234d3f0275124738",
       "version": "72.1",
       "port-version": 1

